package.json

```diff
@@ -67,9 +67,11 @@
   "dependencies": {
     "async": "^2.6.0",
     "boom": "^7.1.1",
+    "debug": "^3.1.0",
     "detect-node": "^2.0.3",
     "hapi": "^16.6.2",
     "hat": "0.0.3",
+    "ipfs-api": "^17.3.0",
```

Comment:
this needs to be moved to devDependencies

Reply:
@dryajov why devDep and not dep?

Reply:
@dryajov  Doesn't `ipfsd-ctl` need `ipfs-api` to control the remote daemon?

Reply:
@richardschneider it does, but we've had issues bundling deps like that before (it was `ipfs` proper in that case, but it applies to `ipfs-api` as well).

Reply:
Is it a circular reference, both need each other?

Reply:
Why does it apply to `ipfs-api` as well? `ipfs` caused issues because it brought in some native dependencies that were causing trouble for ipfs-desktop. It was ok to move `ipfs` to devDep because we moved it to be injected by the user (the other option being go-ipfs-dep). However, `ipfs-api` is always used and doesn't make sense that a user needs to inject as well.

Reply:
IMO, we should not bundle neither `ipfs` not `ipfs-api` and let the consumer bundle their own, because we can't guarantee which version is going to be ultimately used by `ipfsd-ctl` in either case.

> However, ipfs-api is always used and doesn't make sense that a user needs to inject as well.

So is `js-ipfs`, the consuming project might have a different version bundled already. 

I'm fine with bundling `ipfs-api`, I don't think it's going to cause any issues, but allowing the consuming project use their own, IMO gives it more control. It's not a big issue either way, because the consumer can alway explicitly use their own `ipfs-api` version with `IpfsApi(ipfsd.apiAddr)`.

Reply:
Hey guys, I just want some debug logging.  Please make a decision.

Reply:
I'm fine including it 👍

Reply:
> I'm fine with bundling ipfs-api, I don't think it's going to cause any issues, but allowing the consuming project use their own, IMO gives it more control.

What kind of control do you expect? `ipfs-api` is the client library to talk with daemons. 

If I understand correctly, the only reason why we don't pack in `ipfs` and `npm-go-ipfs-dep` by default is due to issues with electron, not because of control.